src/evl.c

```diff
@@ -10,11 +10,18 @@
 
 static int evsrc_tostring(lua_State *L)
 {
-	int ret;
+	int ret, enabled;
 	const char *desc;
+
 	sd_event_source *evsrc = *((sd_event_source**) luaL_checkudata(L, -1, EVSRC_MT));
+
+
+	ret = sd_event_source_get_enabled(evsrc, &enabled);
+	if (ret<0)
+		luaL_error(L, "event_source_get_enabled failed: %s", strerror(-ret));
+
 	ret = sd_event_source_get_description(evsrc, &desc);
-	lua_pushfstring(L, "event_source [%s] %p", (ret<0)?"unkown":desc, evsrc);
+	lua_pushfstring(L, "event_source [%s, %s] %p", (ret<0)?"unkown":desc, (enabled==-1?"oneshot":(enabled==1?"enabled":"disabled")), evsrc);
```

Comment:
I am not proud of this, but ... I wanted to keep a oneliner :grimacing:

Reply:
that's fine :+1: